### PR TITLE
Test create initial element

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Add and test functions for creating the initial elements of the domain

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- [ ] Code has documentation and unit tests
- [ ] Private member variables have a trailing underscore
- [ ] Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- [ ] Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- [ ] File lists in CMake are alphabetical
- [ ] Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- [ ] Mark objects `const` whenever possible
- [ ] Almost always `auto`, except with expression templates, i.e. `DataVector`
- [ ] All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- [ ] Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."

### CODE REVIEW

File: tests/Unit/Domain/DomainTestHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const auto& blocks = domain.blocks();
+  CHECK(blocks.size() == expected_external_boundaries.size());
+  CHECK(blocks.size() == expected_block_neighbors.size());
+  CHECK(blocks.size() == expected_maps.size());
+  for (size_t i = 0; i < blocks.size(); ++i) {
+    const auto& block = blocks[i];
+    CHECK(block.id() == i);
+    CHECK(block.neighbors() == expected_block_neighbors[i]);
+    CHECK(block.external_boundaries() == expected_external_boundaries[i]);
+    CHECK(typeid(*expected_maps[i]) == typeid(block.coordinate_map()));
```
Comment (kidder) [inline tests/Unit/Domain/DomainTestHelpers.cpp:38]: This only checks the types are as expected, not that the maps are equivalent; Will suggests testing them at a few points in the block; Eamonn suggests modifying the are_maps_equal in TestMapHelpers to do this check.

File: tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp [hunk 2]
```diff
@@ -30,6 +30,27 @@ bool are_maps_equal(
...
+    const CoordinateMapBase<SourceFrame, TargetFrame, VolumeDim>& map_two) {
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_real_distribution<> real_dis(-1, 1);
+
+  for (size_t n = 0; n < 10; ++n) {
+    tnsr::I<double, VolumeDim, SourceFrame> source_point{};
+    for (size_t d = 0; d < VolumeDim; ++d) {
+      source_point.get(d) = real_dis(gen);
+      CAPTURE_PRECISE(source_point.get(d));
```
Comment (wthrowe) [inline tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp]: CAPTURE_PRECISE is scoped, so this doesn't ever print anything.  You can CAPTURE_PRECISE an entire tensor, though, so you can just do that after the loop here.

File: tests/Unit/Domain/DomainTestHelpers.hpp [hunk 3]
```diff
@@ -30,3 +33,9 @@ void test_domain_construction(
     const std::vector<std::unique_ptr<
         CoordinateMapBase<Frame::Logical, Frame::Inertial, VolumeDim>>>&
         expected_maps);
+
+// Fraction of the logical volume of a block covered by an element
+// The sum of this over all the elements of a block should be one
+template <size_t VolumeDim>
+boost::rational<size_t> fraction_of_block_volume(
```
Comment (kidder) [inline tests/Unit/Domain/DomainTestHelpers.hpp:40]: noexcept

File: tests/Unit/Domain/Test_InitialElementIds.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  boost::rational<size_t> logical_volume_of_blocks(0);
+  for (const auto& element_id : element_ids) {
+    logical_volume_of_blocks += fraction_of_block_volume(element_id);
+  }
+  CHECK(expected_logical_volume_of_blocks == logical_volume_of_blocks);
+}
+} // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.InitialElementIds", "[Domain][Unit]") {
+  std::vector<std::array<size_t, 1>> initial_refinement_levels_1d{{{2}}, {{3}}};
```
Comment (kidder) [inline tests/Unit/Domain/Test_InitialElementIds.cpp]: const

File: src/Domain/CreateInitialElement.cpp [hunk 5]
```diff
@@ -87,21 +83,15 @@ Element<VolumeDim> create_initial_element(
 }
 
 template Element<1> create_initial_element<1, Frame::Grid>(
```
Comment (kidder) [inline src/Domain/CreateInitialElement.cpp]: use template instantiation macro

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**